src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -1431,3 +1440,142 @@ std::string SotaUptaneClient::secondaryTreehubCredentials() const {
     return "";
   }
 }
+
+LazyTargetsList SotaUptaneClient::allTargets() { return LazyTargetsList(images_repo, storage, uptane_fetcher); }
```

Comment:
I think a call to `updateImagesMeta();` would be really handy. Otherwise, the only code path to trigger that logic is via Uptane specific logic. Alternatively, you could make updateImagesMeta a public method, but I think just call updateImagesMeta might make things easier.

With that fix, things are working just fine for my aktualizr-lite logic

Reply:
@doanac I think for now we'll have to just make it public to keep the interface of `SotaUptaneClient` consistent in that each call does a simple contained thing. What we could do to make it user-friendly is to add TUF API pretty much as `class Aktualizr` represents our Uptane API. That could wrap two calls into one.